x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/sample_logs_input.tsx

```diff
@@ -64,15 +65,34 @@ export const parseJSONArray = (
   return { errorNoArrayFound: true, entries: [], pathToEntries: [] };
 };
 
+/**
+ * Truncates an array of log samples type T and returns a new array.
+ *
+ * Note array cannot be empty since we keep the first element in place.
+ *
+ * @param array - The array to truncate.
+ * @returns The truncated array with no more than MaxLogsSampleRows.
+ * @template T - The type of elements in the array.
+ */
+function _truncateLogSample<T>(array: T[]): T[] {
+  const numElements = Math.min(MaxLogsSampleRows, array.length);
+  partialShuffleArray(array, 1, numElements);
+  return array.slice(0, numElements);
+}
```

Comment:
I have some questions about this function:

- Couldn't we just truncate the logs array first and then shuffle the entire truncated array? It would simplify the implementation.
- It is called _truncateLogSample_, but it does not only truncate the logs but also shuffle the elements. It should be renamed to something like () or the shuffle call extracted from it.
- Why is the function name starting with `_`? It is unusual in Kibana.